skopt/plots.py

```diff
@@ -88,3 +90,245 @@ def plot_convergence(*args, **kwargs):
+    for i in range(space.n_dims):
+        for j in range(space.n_dims):
+            ax_ = ax[i, j]
+
+            if j > i:
+                ax_.axis("off")
+
+            # adjust bounds for every off-diagonal axis
+            if i != j:
+                ax_.set_ylim(*space.dimensions[i].bounds)
+                ax_.set_xlim(*space.dimensions[j].bounds)
+                # keep y axis tick labels for the diagonal
+                if j > 0:
+                    ax_.set_yticklabels([])
+            else:
+                ax_.yaxis.tick_right()
+                ax_.yaxis.set_label_position('right')
+                ax_.yaxis.set_ticks_position('both')
+                ax_.set_ylabel(ylabel)
+
+            ax_.xaxis.set_major_locator(MaxNLocator(6, prune='both'))
+            ax_.yaxis.set_major_locator(MaxNLocator(6, prune='both'))
+
+            if i < space.n_dims - 1:
+                ax_.set_xticklabels([])
+            # bottom row
+            else:
+                [l.set_rotation(45) for l in ax_.get_xticklabels()]
+
+    return ax
+
+
+def partial_dependence(space, model, i, j=None, sample_points=None,
+                       n_samples=100, n_points=40):
+    """Calculate partial dependence of `model` for dimensions `i` and `j`
+
+    The idea is that the partial dependence plot tells us how the
+    value of the dimensions `i` and `j` influence the `model` predictions
+    after we have "averaged out" the influence of all other dimensions.
+
+    Parameters
+    ----------
+    * `space` [`Space`]
+        The parameter space over which the minimization was performed.
+
+    * `model` [XXX]
+        Surrogate model for the objective function.
+
+    * `i` [int]
+        The first dimension for which to calculate the partial dependence.
+
+    * `j` [int, default=None]
+        The second dimension for which to calculate the partial dependence.
+        To calculate the 1D partial dependence on `i` alone set `j=None`.
+
+    * `sample_points` [list of lists, shape=(n_points, n_dims), default=None]
+        Randomly sampled and transformed points to use when averaging
+        the model function at each of the `n_points`.
+
+    * `n_samples` [int, default=100]
+        Number of random samples to use for averaging the model function
+        at each of the `n_points`. Only used when `sample_points=None`.
+
+    * `n_points` [int, default=40]
+        Number of points at which to evaluate the partial dependence
+        along each dimension `i` and `j`.
+
+    Returns
+    -------
+    For 1D partial dependence:
+
+    * `xi`: [np.array]:
+        The points at which the partial dependence was evaluated.
+
+    * `yi`: [np.array]:
+        The value of the model at each point `xi`.
+
+    For 2D partial dependence:
+
+    * `xi`: [np.array]:
+        The points at which the partial dependence was evaluated.
+    * `yi`: [np.array]:
+        The points at which the partial dependence was evaluated.
+    * `zi`: [np.array]:
```

Comment:
Done. Actually it is a list of lists. I think this is ok (but easy to turn into an array if that's what the world wants).